### PR TITLE
feat: get appropriate placeholder for each empty filter field

### PR DESCRIPTION
**Pull request by joaoviana**
Closes: #6383

### Description:

Allow for `placeholder` prop in all filter components.
Display `any value` disabled and empty Input if **filter rule** is `disabled`
Create `getPlaceholderByFilterTypeAndOperator` util to get appropriate filter input placeholder - depending on the filter's type and operator.

Screen recordings:

Boolean

https://github.com/lightdash/lightdash/assets/7611706/6dbe67d9-520a-40f2-9367-30f96aee97c6

Number

https://github.com/lightdash/lightdash/assets/7611706/e2659a49-665e-4754-a265-19eaf120856e

String

https://github.com/lightdash/lightdash/assets/7611706/6986ac86-6c48-4989-b044-5efe22800278

Date

https://github.com/lightdash/lightdash/assets/7611706/13289054-fdb0-4764-a9b6-a9158aff7a0b

NOTE:
Some placeholders won't be visible at first because we provide users with default values in most scenarios. What we intend to do with this work is to help users understand what they should do when they select a new filter. 
For example, a number-type filter starts `empty` by default, so the placeholder should display and help the user know what should be done (same for string filters, where the user can start typing and filtering results and default value is also empty) 
For date filters, we do set default values and have safeguards for incorrect dates, so in most cases the user won't see a placeholder, because the values set are already an indication of what format the value should be in.

### CODE REVIEW

File: packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx [hunk 1]
```diff
@@ -23,6 +23,73 @@ export type FilterInputsProps<T extends ConditionalRule> = {
...
+    filterOperator: FilterOperator,
+) => {
+    switch (filterType) {
+        case FilterType.NUMBER:
+            switch (filterOperator) {
+                case FilterOperator.EQUALS:
+                case FilterOperator.NOT_EQUALS:
+                case FilterOperator.LESS_THAN:
+                case FilterOperator.GREATER_THAN:
+                    return 'Enter Value(s)';
```
Comment (rephus) [inline packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx:37]: lowercase `value` ?

File: packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx [hunk 2]
```diff
@@ -23,6 +23,73 @@ export type FilterInputsProps<T extends ConditionalRule> = {
...
+                case FilterOperator.GREATER_THAN:
+                case FilterOperator.GREATER_THAN_OR_EQUAL:
+                    return 'Select a date';
+                case FilterOperator.IN_THE_PAST:
+                case FilterOperator.NOT_IN_THE_PAST:
+                case FilterOperator.IN_THE_NEXT:
+                case FilterOperator.IN_THE_CURRENT:
+                    return 'Select a time period';
+                case FilterOperator.IN_BETWEEN:
+                    return 'Start date End date';
```
Comment (rephus) [inline packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx:73]: how is this rendered ?

File: packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx [hunk 3]
```diff
@@ -23,6 +23,73 @@ export type FilterInputsProps<T extends ConditionalRule> = {
...
+const getPlaceholderByFilterTypeAndOperator = (
+    filterType: FilterType,
+    filterOperator: FilterOperator,
+) => {
+    switch (filterType) {
+        case FilterType.NUMBER:
+            switch (filterOperator) {
+                case FilterOperator.EQUALS:
+                case FilterOperator.NOT_EQUALS:
+                case FilterOperator.LESS_THAN:
```
Comment (rephus) [inline packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx:35]: This output doesn't match the spect, less_than should be `Enter value`

Comment (rephus) [reply packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx:35]: image

File: packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx [hunk 4]
```diff
@@ -23,6 +23,73 @@ export type FilterInputsProps<T extends ConditionalRule> = {
...
+                    return 'Start date End date';
+                case FilterOperator.NULL:
+                case FilterOperator.NOT_NULL:
+                default:
+                    return '';
+            }
+        case FilterType.BOOLEAN:
+            switch (filterOperator) {
+                case FilterOperator.EQUALS:
+                    return 'True or False';
```
Comment (rephus) [inline packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx:82]: Redudant ? There are no placefolders on booleans 

image

File: packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx (hunk 2 above)
Comment (rephus) [reply packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx:73]: I think this doesn't do anything

File: packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx [hunk 5]
```diff
@@ -23,6 +23,73 @@ export type FilterInputsProps<T extends ConditionalRule> = {
...
+                case FilterOperator.LESS_THAN:
+                case FilterOperator.LESS_THAN_OR_EQUAL:
+                case FilterOperator.GREATER_THAN:
+                case FilterOperator.GREATER_THAN_OR_EQUAL:
+                    return 'Select a date';
+                case FilterOperator.IN_THE_PAST:
+                case FilterOperator.NOT_IN_THE_PAST:
+                case FilterOperator.IN_THE_NEXT:
+                case FilterOperator.IN_THE_CURRENT:
+                    return 'Select a time period';
```
Comment (rephus) [inline packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx:71]: Not needed

image

**Review by rephus — COMMENTED** [review]
I think date filters are not working as expected

File: packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx [hunk 6]
```diff
@@ -112,6 +186,7 @@ const DefaultFilterInputs = <T extends ConditionalRule>({
                     className={disabled ? 'disabled-filter' : ''}
                     disabled={disabled}
                     fill
+                    placeholder={placeholder}
```
Comment (IrakliJani) [inline packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx:189]: @joaoviana  are you applying this placeholders only on DefaultFilterInputs or is there going to be another PR for for example DateFilterInputs?

File: packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx (hunk 1 above)
Comment (joaoviana) [reply packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx]: done 👍 , thanks for checking! also updated the issue description

File: packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx (hunk 2 above)
Comment (joaoviana) [reply packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx]: Going to investigate the date filters now, I think more work can be done on a separate PR though

File: packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx (hunk 3 above)
Comment (joaoviana) [reply packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx]: Thanks for checking! changed it and added screen recording ^

File: packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx (hunk 5 above)
Comment (joaoviana) [reply packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx]: Will be needed when filters are allowed to be disabled in the UI, i.e. adding the filter, operator, but no value is set, yet

File: packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx (hunk 6 above)
Comment (joaoviana) [reply packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx:123]: Thanks for raising this @IrakliJani, I'll be doing this on this PR!

**Comment by joaoviana** [comment]
moved to draft to work on date-type filters

File: packages/frontend/src/components/DashboardFilter/FilterConfiguration/index.tsx [hunk 7]
```diff
@@ -90,14 +89,12 @@ const FilterConfiguration: FC<Props> = ({
...
-                    operator: operator,
-                }),
-            );
+            setInternalFilterRule((prevState) => ({
+                ...prevState,
+                operator,
+            }));
         },
-        [field],
+        [],
```
Comment (joaoviana) [inline packages/frontend/src/components/DashboardFilter/FilterConfiguration/index.tsx:97]: check this

File: packages/frontend/src/components/DashboardFilter/FilterConfiguration/FilterSettings.tsx [hunk 8]
```diff
@@ -35,22 +37,48 @@ const FilterSettings: FC<FilterSettingsProps> = ({
...
+                ? filterRule.values
+                : getFilterRuleWithDefaultValue(field, filterRule).values;
+        }
+
+        onChangeFilterRule({
+            ...filterRule,
+            disabled: !isToggled,
+            values: filterValues,
+        });
+    };
```
Comment (joaoviana) [inline packages/frontend/src/components/DashboardFilter/FilterConfiguration/FilterSettings.tsx:57]: don't think we need most of this

File: packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx (hunk 4 above)
Comment (joaoviana) [reply packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx]: Ignore the above ^ removed as it's not necessary, thanks!

**Review by IrakliJani — APPROVED** [review]

File: packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx (hunk 5 above)
Comment (joaoviana) [reply packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx]: exactly, not needed as you said @rephus , thanks! added a comment explaining as well

**Comment by joaoviana** [comment]
> Do we need placeholder for this date ?
> 
> image

And done! 
image

all screen recordings in description
